src/EditorFeatures/Core/FindUsages/IDefinitionsAndReferencesFactory.cs

```diff
@@ -144,7 +144,7 @@ internal static class DefinitionItemExtensions
             }
 
             return DefinitionItem.Create(
-                tags, displayParts, sourceLocations.ToImmutableAndFree(),
+                tags, displayParts, sourceLocations.ToImmutable(),
```

Comment:
jsut to check, but did you validate all codepaths to ensure that through nothin else do we return ToImmutableAndFRee?  (this applies to all your changes).

Reply:
I'm not sure I follow. There were certain cases (e.g. the one above) where there are were return paths that didn't Free (but that also don't return the builder / its underlying array (AFAIK) so there should be no issue).

Reply:
I tried to make sure no double frees were induced, if that's what you're asking about.

Reply:
> I tried to make sure no double frees were induced, if that's what you're asking about.

Yes.

Consider, for example, something like this:


```c#
var builder = ArrayBuilder.GetInstance();
// 50 lines of code.
if (whatever) return builder.ToImmutableAndFree();
// 50 lines of code
return builder.ToImmutableAndFree();
```

If you add the `using`, you need to update both return statements.  In PR form this is hard for the reviewer to determine as it means examinign all these methods themselves.  So i'm asking if you've appropriately audited to make sure that you've updated all the places that need to be updated :)